ui/app/src/state/actions/system.ts

```diff
@@ -126,5 +126,12 @@ export const fetchGlobalMenuFailed = /*#__PURE__*/ createAction('FETCH_GLOBAL_ME
 export const blockUI = /*#__PURE__*/ createAction<Partial<UIBlockerStateProps>>('BLOCK_UI');
 export const unblockUI = /*#__PURE__*/ createAction('UNBLOCK_UI');
 
-export const openSiteSocket = /*#__PURE__*/ createAction<{ site: string; xsrfToken: string }>('OPEN_SITE_SOCKET');
+export const openSiteSocket = /*#__PURE__*/ createAction<{ site?: string; xsrfToken: string }>('OPEN_SITE_SOCKET');
```

Comment:
Let's stay out of trouble: keep openSiteSocket dedicated to site topic. The SharedWorker can connect/disconnect from global topic based on login/timeout, doesn't need an instruction from the UI since its not conditional to what site is selected.
The only issue might be the xsrf token but if push comes to shove, create a devoted action for global socket topic.
```suggestion
export const openSiteSocket = /*#__PURE__*/ createAction<{ site: string; xsrfToken: string }>('OPEN_SITE_SOCKET');
```